web/src/routes/(user)/search/+page.server.ts

```diff
@@ -0,0 +1,26 @@
+import { redirect } from '@sveltejs/kit';
+import type { PageServerLoad } from './$types';
+
+export const load = (async ({ locals, parent, url }) => {
+	const { user } = await parent();
+	if (!user) {
+		throw redirect(302, '/auth/login');
+	}
+
+	const term = url.searchParams.get('q') || undefined;
+
+	const { data: results } = await locals.api.searchApi.search(
+		term,
+		undefined,
+		undefined,
+		undefined,
+		undefined,
+		undefined,
+		undefined,
+		undefined,
+		undefined,
+		undefined,
```

Comment:
What the heck is going on here? Lol

Reply:
For some reason query params each get their own argument spot lol, but the last argument also allows you to pass a bunch of extra axios options, including params. I decided to pass everything from the original url instead of mapping then individuality.